### ISSUE TITLE
Question about `only` option in TestDefinition

### ISSUE THREAD

**Issue by kt3k** [denoland/deno#17262]
related https://github.com/denoland/deno_std/issues/2979

`only` option looks scoped to a file. (i.e. if one test case in `a_test.ts` has `only: true`, then other test cases in `a_test.ts` are skipped, but if there are other test files (`b_test.ts`), then those test cases in other test files are still executed with `deno test` command).

Is this the intended behavior?

**Comment by kt3k** [comment]
`deno test` with the following files:

a_test.ts
```ts
Deno.test("A 1", () => {});
Deno.test("A 2", { only: true }, () => {});
```

b_test.ts
```ts
Deno.test("B", () => {});
```

This results with the below executions:
```shellsession
$ deno test    
running 1 test from ./a_test.ts
A 2 ... ok (7ms)
running 1 test from ./b_test.ts
B ... ok (6ms)

ok | 2 passed | 0 failed | 1 filtered out (68ms)
```

The case `A 1` is filtered out, but `B` is not

**Comment by bartlomieju** [comment]
Good question - I'm not sure if we can figure out upfront which test cases from which files we need to run because test steps with `.only` option might be deeply nested. @dsherret any idea?

**Comment by KyleJune** [comment]
I know that other test runners like jasmine won't run tests from other files if one file has a focused test. This has been a bit of a pain point for me working with deno because you have to pass a file argument to the test command to be able to focus a file. I use many flags and have a shorthand in my deno task list. I end up needing to modify it if I want to focus a specific file that has the test I want to focus.

I think it would be a good change to make it so that if any file has focused tests, only those files will run.

**Comment by bartlomieju** [comment]
Yes, this is intentional. `only` is file-scoped, not global, and that falls out of the test runner architecture rather than being a special rule.

Each test file is loaded and run in its own worker isolate, and they run in parallel (`test_specifiers` spawns one worker per specifier in `cli/tools/test/mod.rs`). The `only` filtering happens in `compute_tests_to_run` (`cli/tools/test/mod.rs`), which only ever sees the tests registered by that single file. There is no global phase that collects every test across all files before deciding what to run, so `only` cannot suppress tests in other files.

The one piece that is global is the failure: if any file used `only`, the `used_only` flag is aggregated in `report_tests` and the whole run exits with `Test failed because the "only" option was used`. So `only` narrows execution within its own file, and globally it just turns the run red so it cannot be committed by accident.

Current behavior confirmed on canary:

```
a_test.ts: Deno.test({ name: "a-only", only: true, ... }); Deno.test({ name: "a-other", ... });
b_test.ts: Deno.test({ name: "b-1", ... });               Deno.test({ name: "b-2", ... });

$ deno test
running 1 test from ./a_test.ts   # a-only runs, a-other filtered out
running 2 tests from ./b_test.ts  # both b-1 and b-2 still run
ok | 3 passed | 0 failed | 1 filtered out
error: Test failed because the "only" option was used
```

Closing as working as intended.
